module/os/linux/zfs/zpl_super.c

```diff
@@ -375,12 +384,20 @@ zpl_prune_sb(uint64_t nr_to_scan, void *arg)
 	struct super_block *sb = (struct super_block *)arg;
 	int objects = 0;
 
-	(void) -zfs_prune(sb, nr_to_scan, &objects);
+	if (atomic_inc_not_zero(&sb->s_active)) {
```

Comment:
would like a comment about what we're trying to avoid here (i.e. from the PR description)